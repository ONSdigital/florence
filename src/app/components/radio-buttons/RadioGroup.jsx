import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Radio from './RadioButton';

const propTypes = {
    radioData: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired
    })).isRequired,
    legend: PropTypes.string,
    groupName: PropTypes.string.isRequired,
    onChange: PropTypes.func,
    selectedValue: PropTypes.string,
    inline: PropTypes.bool
};

export default class RadioGroup extends Component {
    constructor(props) {
        super(props);

        //this.state = {selectedValue: ""};

        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        if (this.props.onChange) {
            this.props.onChange(event);
        }
    }

    render() {
        const radioData = this.props.radioData;
        const groupName = this.props.groupName;
        const selectedValue = this.props.selectedValue;

        return (
            <fieldset className="fieldset">
                {this.props.legend ? <legend>{this.props.legend}</legend> : ""}
                {
                    radioData.map((radio, index) => {
                        return <Radio inline={this.props.inline} key={index} {...radio} group={groupName} onChange={this.handleChange} checked={selectedValue === radio.value}/>
                    })
                }
            </fieldset>
        )
    }
}

RadioGroup.propTypes = propTypes;