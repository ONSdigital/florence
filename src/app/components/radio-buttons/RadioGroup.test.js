import React from 'react';
import RadioGroup from './RadioGroup';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const defaultProps = {
    radioData: [
        {
            id: "test-radio1",
            value: "test-value1",
            label: "Test radio 1",
        },
        {
            id: "test-radio2",
            value: "test-value2",
            label: "Test radio 2",
        }
    ],
    legend: "Select a test radio",
    groupName: "test-group",
    onChange: ()=>{},
    selectedValue: "test-value1",
    inline: false
};

const mockedEvent = {id: "test-radio2", checked: true, value: "test-value2"};

test("Radio button matches stored snapshot", () => {

    const component = renderer.create(
        <RadioGroup {...defaultProps} />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test("Handle change passes correct value caller onChange method", () => {
    let passedToParent = {};
    const mockedOnChange = (event) => {passedToParent = event;};
    const component = shallow(
        <RadioGroup {...defaultProps} checked={false} onChange={mockedOnChange}/>
    );
    //expect(component.update().state().selectedValue).toBe("test-value1");
    component.instance().handleChange(mockedEvent);
    //expect(component.update().state().selectedValue).toBe("test-value2");
    expect(passedToParent).toMatchObject({
        id: "test-radio2",
        checked: true,
        value: "test-value2"
    });
});


