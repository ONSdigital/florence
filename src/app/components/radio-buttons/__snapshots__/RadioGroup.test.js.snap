// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio button matches stored snapshot 1`] = `
<fieldset
  className="fieldset"
>
  <legend
    className="fieldset__legend"
  >
    Select a test radio
  </legend>
  <div
    className="radio"
  >
    <input
      checked={true}
      className="radio__input selected"
      id="test-radio1"
      name="test-group"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="radio"
      value="test-value1"
    />
    <label
      className="radio__label radio__label--selected"
      htmlFor="test-radio1"
    >
      Test radio 1
       
    </label>
  </div>
  <div
    className="radio"
  >
    <input
      checked={false}
      className="radio__input"
      id="test-radio2"
      name="test-group"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="radio"
      value="test-value2"
    />
    <label
      className="radio__label"
      htmlFor="test-radio2"
    >
      Test radio 2
       
    </label>
  </div>
</fieldset>
`;
